### PR TITLE
Pull kubeflow pipelines manifest from upstream using kpt

### PR DESCRIPTION
**Pull request by Bobgy**
**NOTE**: the branch is based on https://github.com/kubeflow/manifests/pull/1241, so please merge it first.
To review this PR, please look at https://github.com/kubeflow/manifests/pull/1242/files/1c43da3eeeb2883e4072bd0681c73fc9aeca2e75..3fa9d2194ed715d2c9c53014207bf47622315ce6

**Which issue is resolved by this Pull Request:**
Resolves https://github.com/kubeflow/pipelines/issues/3962

**Description of your changes:**
Pull kubeflow pipelines manifest from upstream using kpt

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate-changed-only`
    3. `make test`

### CODE REVIEW

File: hack/pull_kfp_upstream.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
```
Comment (rmgogogo) [inline hack/pull_kfp_upstream.sh:1]: a doc for the pull action?

Comment (Bobgy) [reply hack/pull_kfp_upstream.sh:1]: just use it `./hack/pull_kfp_upstream.sh`

Anything else we'd want to document?

**Comment by Bobgy** [comment]
Already merged in https://github.com/kubeflow/manifests/pull/1247.
/close
@rmgogogo Do you have further concerns? I can create a new PR with requested changes.

**Pull request closed without merging**